--- src/elf_loader/src/loader/src/loader.c
+++ src/elf_loader/src/loader/src/loader.c
@@ -385,15 +385,18 @@
		SifExitRpc();

		if (is_hdd_partition_context(partition_context)) {
-			while(!SifIopReset("", 0)){};
+			while(!SifIopReset("rom0:UDNL rom0:EELOADCNF", 0)){};
			while (!SifIopSync()) {};

			SET_GS_BGCOLOUR(ORANGE_BG);

			/* Re-initialize RPC temporarily to load the necessary modules */
			SifInitRpc(0);
			SifLoadFileInit();
			SifLoadModule("rom0:SIO2MAN", 0, NULL);
+			SifLoadModule("rom0:CDVDFSV", 0, NULL);
+			SifLoadModule("rom0:CDVDMAN", 0, NULL);
			SifLoadModule("rom0:MCMAN", 0, NULL);
			SifLoadModule("rom0:MCSERV", 0, NULL);
+			SifLoadModule("rom0:PADMAN", 0, NULL);
