--- bin/POPSLDR/system.lua
+++ bin/POPSLDR/system.lua
@@ -1181,9 +1181,8 @@
       end
       local app_part = string.match(app_dir, "^(hdd0:.-):")
-      local app_slot = string.match(app_dir, ":pfs(%d*):")
-      if app_slot == "" then app_slot = "0" end
-      if app_part and app_slot == slot_str then
-          return app_part .. ":"
+      if app_part then
+        return app_part .. ":"
       end
     end
   end
