--- src/elf_loader/src/loader/src/loader.c
+++ src/elf_loader/src/loader/src/loader.c
@@ -346,24 +346,6 @@
		SET_GS_BGCOLOUR(YELLOW_BG);

		/* Properly teardown any EE-side RPC clients before ExecPS2 */
-		if (is_hdd_partition_context(partition_context)) {
-			/* We must wipe the IOP so the dirty HDD/fileXio modules are gone before ExecPS2. */
-			while (!SifIopReset("rom0:UDNL rom0:EELOADCNF", 0)) {
-			}
-			while (!SifIopSync()) {
-			}
-
-			SifLoadFileExit();
-			SifExitRpc();
-			SifExitCmd();
-
-			SET_GS_BGCOLOUR(ORANGE_BG);
-
-			/* Re-initialize RPC temporarily to load the necessary modules */
-			SifInitRpc(0);
-			SifLoadFileInit();
-			SifLoadModule("rom0:SIO2MAN", 0, NULL);
-			SifLoadModule("rom0:CDVDFSV", 0, NULL);
-			SifLoadModule("rom0:CDVDMAN", 0, NULL);
-			SifLoadModule("rom0:MCMAN", 0, NULL);
-			SifLoadModule("rom0:MCSERV", 0, NULL);
-			SifLoadModule("rom0:PADMAN", 0, NULL);
-
-			/* Teardown again before ExecPS2 */
-			SifLoadFileExit();
-			/* DO NOT teardown SifExitRpc or SifExitCmd! USB POPSTARTER inherits
-			   them active from LoadELFFromFileExecPS2, so HDD must too! */
-		} else {
-			SifLoadFileExit();
-		}

		SET_GS_BGCOLOUR(BROWN_BG);
